skale/manager_client.py

```diff
@@ -1,26 +1,41 @@
+import os
 import logging
 
 from web3 import Web3
 import skale.contracts as contracts
 from skale.contracts_info import CONTRACTS_INFO
+from skale.transactions.manager import TransactionsManager
+from skale.transactions.tools import send_to_tx_manager, sign_and_send
+from skale.utils.account_tools import create_wallet
 from skale.utils.helper import get_abi
 from skale.utils.web3_utils import get_provider
-from skale.transactions_manager import TransactionsManager
 
 logger = logging.getLogger(__name__)
 
 
+class EmptyPrivateKey(Exception):
+    pass
+
+
 class Skale:
-    def __init__(self, endpoint, abi_filepath, transactions_manager_url=None):
+    def __init__(self, endpoint, abi_filepath, wallet=None,
+                 tx_manager_url=None):
         logger.info(f'Init skale-py, connecting to {endpoint}')
         provider = get_provider(endpoint)
         self.web3 = Web3(provider)
         self.abi = get_abi(abi_filepath)
         self.__contracts = {}
         self.__contracts_info = {}
         self.nonces = {}
-        if transactions_manager_url:
-            self.transactions_manager = TransactionsManager(transactions_manager_url)
+        if tx_manager_url:
+            self.tx_manager = TransactionsManager(tx_manager_url)
+        else:
+            private_key = os.getenv('ETH_PRIVATE_KEY')
+            if not private_key:
+                raise EmptyPrivateKey()
+            self.wallet = wallet or create_wallet(web3=self.web3,
```

Comment:
it looks like wallet object is not inited if we're using `tx_manager`, but in `manager` contract we're using `self.skale.wallet.public_key`